src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -497,7 +495,7 @@ public static String toUpperCase(String str, byte[] value, Locale locale) {
         // Now check if there are any characters that need to be changed, or are surrogate
         for (first = 0 ; first < len; first++ ) {
             int cp = value[first] & 0xff;
-            if (cp != CharacterDataLatin1.instance.toUpperCaseEx(cp)) {   // no need to check Character.ERROR
+            if (cp >= 'a' && (cp <= 'z' || cp == 181 || (cp >= 223 && cp != 247))) {
```

Comment:
The following would be consistent with your other changes:
```java
if (CharacterDataLatin1.instance.isLowerCase(value[first] & 0xff)) {
```

Reply:
When the value of cp is 170 or 186, the result of using isLowerCase is not expected, and it also cause build fail.

Reply:
Should we define a new method in `CharacterDataLain1` for such a dedicated check?